src/github/utils/config.ts

```diff
@@ -1,15 +1,20 @@
 import { Value } from "@sinclair/typebox/value";
 import { GitHubContext } from "../github-context";
 import YAML from "yaml";
-import { Config, configSchema } from "../types/config";
 import { expressionRegex } from "../types/plugin";
+import { configSchema, PluginConfiguration } from "../types/plugin-configuration";
 import { eventNames } from "../types/webhook-events";
+import { BotConfig, generateConfiguration } from "@ubiquibot/configuration";
 
-const UBIQUIBOT_CONFIG_FULL_PATH = ".github/ubiquibot-config.yml";
+const UBIQUIBOT_CONFIG_FULL_PATH = ".github/.ubiquibot-config.yml";
 
-export async function getConfig(context: GitHubContext): Promise<Config | null> {
+export async function getConfig(context: GitHubContext): Promise<BotConfig | null> {
```

Comment:
looking at the [BotConfig from @ubiquibot/configuration](https://github.com/ubiquibot/configuration/blob/development/src/types/configuration-types.ts) it looks like that is the old config which is not used anymore

Reply:
It also contains the new configuration with the plugins
https://github.com/ubiquibot/configuration/blob/development/src/types/configuration-types.ts#L118
I also have this PR addressing the config changes with a proper default config
https://github.com/ubiquibot/configuration/pull/14/

I could remove all the unused stuff in the aforementioned PR for sure. Are only `plugins` relevant at this point?

Reply:
yes basically only `plugins` property is relevant to the kernel and other properties were from the previous bot where all functionality was in one codebase but now all those properties will be plugin-specific so they will be defined and handled by plugins themselves

Reply:
@whilefoo Okay noted, then I will cleanup the https://github.com/ubiquibot/configuration accordingly

Reply:
Should be solved by https://github.com/ubiquibot/configuration/pull/17